k8sdeps/kunstruct/factory.go

```diff
@@ -101,8 +101,9 @@ func (kf *KunstructuredFactoryImpl) Set(fs fs.FileSystem, ldr ifc.Loader) {
 }
 
 // validate validates that u has kind and name
+// except for kind `List`, which doesn't require a name
 func (kf *KunstructuredFactoryImpl) validate(u unstructured.Unstructured) error {
-	if u.GetName() == "" {
+	if u.GetName() == "" && u.GetKind() != "List" {
```

Comment:
Can you also reorganize the if statements here? Put the checks for `Kind` together.
```
 	if u.GetName() == "" && u.GetKind() != "List" {
 		return fmt.Errorf("missing metadata.name in object %v", u)	 		
    }
	if u.GetKind() == "" {
 		return fmt.Errorf("missing kind in object %v", u)	 		 	
    }
```

Reply:
Do you mean something like this?

note: this changed the priority of the two errors

```go
        kind := u.GetKind()
        if kind == "" {
                return fmt.Errorf("missing kind in object %v", u)
        } else if kind == "List" {
                return nil
        }
        if u.GetName() == "" {
                return fmt.Errorf("missing metadata.name in object %v", u)
        }
        return nil
```

Reply:
Yep, this looks good.